kore-rpc-types/src/Kore/JsonRpc/Types.hs

```diff
@@ -177,6 +178,48 @@ data GetModelResult = GetModelResult
         (FromJSON, ToJSON)
         via CustomJSON '[OmitNothingFields, FieldLabelModifier '[CamelToKebab]] GetModelResult
 
+data SimplifyImplicationResult = SimplifyImplicationResult
+    { validity :: ImplicationValidityResult
+    , substitution :: Maybe KoreJson
+    , logs :: Maybe [LogEntry]
+    }
+    deriving stock (Generic, Show, Eq)
+    deriving
+        (FromJSON, ToJSON)
+        via CustomJSON '[OmitNothingFields, FieldLabelModifier '[CamelToKebab]] SimplifyImplicationResult
+
+data ImplicationValidityResult
+    = -- | implication is valid
+      ImplicationValid
+    | -- | implication is invalid, explains why
+      ImplicationInvalid ImplicationInvalidReason
+    | -- | implication is unknown, explains why
+      ImplicationUnknown ImplicationUnknownReason
+    deriving stock (Generic, Show, Eq)
+    deriving
+        (FromJSON, ToJSON)
+        via CustomJSON '[] ImplicationValidityResult
```

Comment:
could use the same `via CustomJSON '[FieldLabelModifier '[CamelToKebab]]` as above. I think this is the naming convention in pyk?